src/Lesson2/ArrayTest.java

```diff
@@ -0,0 +1,98 @@
+
+import java.util.Arrays;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+public class ArrayTest {
+    static long result;
+    public static void main(String[] args) {
+//        testArray();
+//        testDynamicArray();
+        testSort();
+    }
+
+    private static void testSort() {
+        Random rand = new Random();
+        Array<Integer> data = new ArrayImpl<>();
+        Array<Integer> data2 = new ArrayImpl<>();
+        Array<Integer> data3 = new ArrayImpl<>();
+        Array<Integer> data4 = new ArrayImpl<>();
+
+        for (int i = 0; i < 10000; i++) {
+            int temp = rand.nextInt();
+            data.add(temp);
+            data2.add(temp);
+            data3.add(temp);
+            data4.add(temp);
+        }
+
+        long start = System.nanoTime();
+        data.sortBubble(true);
+        long finish = System.nanoTime();
+        System.out.println("Время сортировки пузырьком " + time(start, finish) + "мс");
+        result = 0;
+
+        start = System.nanoTime();
+        data2.sortBubble(false);
+        finish = System.nanoTime();
+        System.out.println("Время сортировки пузырьком без проверки отсортированной части " + time(start, finish) + "мс");
+        result = 0;
+
+        start = System.nanoTime();
+        data3.sortSelect();
+        finish = System.nanoTime();
+        System.out.println("Время сортировки выбором " + time(start, finish) + "мс");
+
+        start = System.nanoTime();
+        data4.sortInsert();
+        finish = System.nanoTime();
+        System.out.println("Время сортировки вставкой " + time(start, finish) + "мс");
+    }
+
+    private static void testDynamicArray() {
+//        Array<Integer> data = new ArrayImpl<>();
+        Array<Integer> data = new SortedArrayImpl<>();
+        data.add(5);
+        data.add(4);
+        data.add(3);
+        data.add(2);
+        data.add(1);
+
+        data.display();
+
+        data.remove(Integer.valueOf(3));
+//        data.remove(2);
+        data.display();
+
+        System.out.println("data[1] = " + data.get(1));
+    }
+
+    private static void testArray() {
+        //        Integer[] data = new Integer[5];
+//        Integer[] data = new Integer[] {1, 2, 3, 4, 5};
+        Integer[] data = {1, 2, 3, 4, 5};
+        data[2] = 777;
+
+//        Integer[] data2 = new Integer[data.length * 2];
+//        System.arraycopy(data, 0, data2, 0, data.length);
+        data = Arrays.copyOf(data, data.length * 2);
+
+//        data2[5] = 888;
+        data[5] = 888;
+
+//        for (Integer value : data2) {
+        for (Integer value : data) {
+            System.out.println(value);
+        }
+        new Random().nextInt();
+        long start = System.nanoTime();
+        TimeUnit.NANOSECONDS.toMillis(start);
+//        System.out.println(data2[4]);
+    }
+
+    private static long time(long start, long finish){
+        result = finish - start;
+        return TimeUnit.NANOSECONDS.toMillis(result);
```

Comment:
можно избавиться от статического поля, чтобы его не обнулять постоянно
```suggestion
        return TimeUnit.NANOSECONDS.toMillis( finish - start);
```